add documentation for StateStore Query API (#1874)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/state-management/state-management-overview.md

- Distributed concurrency and data consistency
- Bulk [CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete) operations

> - Querying the data

Reviewer: ```suggestion
- Querying state data
```

### Actor state
Transactional state stores can be used to store actor state. To specify which state store to be used for actors, specify value of property `actorStateStore` as `true` in the metadata section of the state store component. Actors state is stored with a specific scheme in transactional state stores, which allows for consistent querying. Only a single state store component can be used as the statestore for all actors. Read the [API reference]({{< ref state_api.md >}}) to learn more about state stores for actors and the [actors API reference]({{< ref actors_api.md >}})

> ### State query

Reviewer: ```suggestion
### Querying state 
```

There are two ways to query the state:

>  * using [state query API]({{< ref "#state-query-api" >}}) implemented in Dapr runtime,

Reviewer: ```suggestion
 * using the [state query API]({{< ref "#state-query-api" >}}) provided in Dapr runtime,
```

 * querying state store [directly]({{< ref "#query-state-store-directly" >}}).

#### State Query API

{{% alert title="alpha" color="warning" %}}
This API is in **alpha** stage.
{{% /alert %}}

> The State Query API provides unified way of querying state stores regardless of underlying databases.

Reviewer: ```suggestion
The state query API provides a way of querying state stores regardless of underlying database or storage technology.
```

> The state query API provides a way of querying state stores regardless of underlying database or storage technology.

Reviewer: ```suggestion
The state query API provides a way of querying  the key/value data saved using the state management in state stores regardless of underlying database or storage technology, and is an optional API. Using the state management query API you can filter, sort and paginate the key/value data.
```

> This is an optional API. The implementation is split between the Dapr runtime and individual state store components.

Reviewer: ```suggestion

```
